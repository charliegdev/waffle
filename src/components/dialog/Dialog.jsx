import React, { Component } from "react";
import PropTypes from "prop-types";
import { Dialog as EvergreenDialog, Pane } from "evergreen-ui";

class Dialog extends Component {
  static propTypes = {
    children: PropTypes.node.isRequired,
    confirmLabel: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    trigger: PropTypes.node.isRequired
  };

  constructor(props) {
    super(props);

    this.state = {
      isShown: false
    };

    this.close = this.close.bind(this);
    this.open = this.open.bind(this);
  }

  close() {
    this.setState({ isShown: false });
  }

  open() {
    this.setState({ isShown: true });
  }

  render() {
    return (
      <div>
        <Pane>
          <EvergreenDialog
            confirmLabel={this.props.confirmLabel}
            isShown={this.state.isShown}
            title={this.props.title}
            onCloseComplete={this.close}
          >
            {this.props.children}
          </EvergreenDialog>
        </Pane>
        <div onClick={this.open}>{this.props.trigger}</div>
      </div>
    );
  }
}

export default Dialog;
