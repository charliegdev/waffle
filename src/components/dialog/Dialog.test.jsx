import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import { Button } from "evergreen-ui";
import Dialog from "./Dialog";

describe("Dialog", () => {
  let dialog;

  beforeEach(() => {
    dialog = (
      <Dialog
        confirmLabel="OK"
        onConfirm={() => undefined}
        title="Sample Dialog"
        trigger={<Button>Open Dialog</Button>}
      >
        <p>Sample Dialog content</p>
      </Dialog>
    );
  });

  test("Dialog renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(dialog, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  test("Dialog renders the same snapshot", () => {
    const component = renderer.create(dialog);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
