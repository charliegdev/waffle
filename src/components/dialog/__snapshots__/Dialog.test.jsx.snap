// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dialog Dialog renders the same snapshot 1`] = `
<div>
  <div
    className="ðŸ“¦box-szg_border-box css-nil"
  />
  <div
    onClick={[Function]}
  >
    <button
      className="css-69cngj ðŸ“¦fnt-fam_b77syt ðŸ“¦mt_0px ðŸ“¦fnt-sze_12px ðŸ“¦f-wght_500 ðŸ“¦ln-ht_32px ðŸ“¦ltr-spc_0 ðŸ“¦btrr_3px ðŸ“¦bbrr_3px ðŸ“¦btlr_3px ðŸ“¦bblr_3px ðŸ“¦pt_0px ðŸ“¦pb_0px ðŸ“¦pr_16px ðŸ“¦pl_16px ðŸ“¦ml_0px ðŸ“¦mr_0px ðŸ“¦mb_0px ðŸ“¦h_32px ðŸ“¦pst_relative ðŸ“¦dspl_inline-flex ðŸ“¦algn-itms_center ðŸ“¦flx-wrap_nowrap ðŸ“¦box-szg_border-box"
    >
      Open Dialog
    </button>
  </div>
</div>
`;
